python/core/__init__.py.in

```diff
@@ -447,3 +450,35 @@ QgsRasterTransparency.TransparentThreeValuePixel.percentTransparent = property(g
 QgsRasterTransparency.TransparentThreeValuePixel.percentTransparent = QgsRasterTransparency.TransparentThreeValuePixel.percentTransparent.setter(set_pixel_transparency)
 QgsRasterTransparency.TransparentSingleValuePixel.percentTransparent = property(get_pixel_transparency)
 QgsRasterTransparency.TransparentSingleValuePixel.percentTransparent = QgsRasterTransparency.TransparentSingleValuePixel.percentTransparent.setter(set_pixel_transparency)
+
+
+def qgis_data_type_to_numeric_data_type(dataType: Qgis.DataType) -> Optional[numpy.dtype]:
+   qgis_to_numpy_dtype_dict = {
+    Qgis.DataType.UnknownDataType: None,
+    Qgis.DataType.Byte: numpy.byte,
+    Qgis.DataType.Int8: numpy.int8,
+    Qgis.DataType.UInt16: numpy.uint16,
+    Qgis.DataType.Int16: numpy.int16,
+    Qgis.DataType.UInt32: numpy.uint32,
+    Qgis.DataType.Int32: numpy.int32,
+    Qgis.DataType.Float32: numpy.float32,
+    Qgis.DataType.Float64: numpy.float64,
+    Qgis.DataType.CInt16: numpy.complex64,  # NumPy does not have complex int16, using complex64 as closest alternative
+    Qgis.DataType.CInt32: numpy.complex64,  # NumPy does not have complex int32, using complex64 as closest alternative
```

Comment:
that will likely result in incorrect / garbage values. Better mapping to None and erroring out

Reply:
:+1:  I'd go for raising a ValueError "The raster block datatype is not compatible with numpy arrays"